website/static/js/file-browser.js

```diff
@@ -0,0 +1,284 @@
+        ];
+        self.filesData = m.prop($osf.apiV2Url(self.collections[0].path, self.collections[0].pathOptions));
+
+        self.breadcrumbs = m.prop([
+            { label : 'All My Projects', url : 'http://localhost:8000/v2/users/me/nodes/?filter%5Bregistration%5D=false', type : 'collection'}
+        ]);
+        self.nameFilters = [
+            { label : 'Caner Uguz', id : '8q36f', type : 'filter'}
+        ];
+        self.tagFilters = [
+            { tag : 'something', type : 'filter'}
+        ];
+
+        self.updateFilesData = function(linkObject) {
+            linkObject.link = self.generateLinks(linkObject);
+            if (linkObject.link !== self.filesData()) {
+                self.filesData(linkObject.link);
+                self.isLoadedUrl = false; // check if in fact changed
+                self.updateBreadcrumbs(linkObject);
+            }
+        };
+
+        // INFORMATION PANEL
+        self.selected = m.prop([]);
+        self.updateSelected = function(selectedList){
+            self.selected(selectedList);
+            console.log(self.selected());
+        }.bind(self);
+
+
+        // COLLECTIONS PANEL
+        self.activeCollection = m.prop(1);
+        self.updateCollection = function(coll) {
+            self.activeCollection(coll.id);
+            var linkObject = {
+                type : 'collection',
+                data : coll,
+                label : coll.label,
+                id : coll.id
+            };
+            self.updateFilesData(linkObject);
+        };
+
+
+        // USER FILTER
+        self.activeUser = m.prop(1);
+        self.updateUserFilter = function(user) {
+            self.activeUser(user.id);
+            var linkObject = {
+                type : 'user',
+                data : user,
+                label : user.label,
+                id : user.id
+            };
+            self.updateFilesData(linkObject);
+        };
+
+        // Refresh the Grid
+        self.updateList = function(element, isInit, context){
+            if(!self.isLoadedUrl) {
+                var el = element || document.getElementById('pOrganizer');
+                m.mount(el, m.component( ProjectOrganizer, { filesData : self.filesData, updateSelected : self.updateSelected, updateFilesData : self.updateFilesData}));
+                self.isLoadedUrl = true;
+            }
+        }.bind(self);
+
+        // BREADCRUMBS
+        self.updateBreadcrumbs = function(linkObject){
+            var crumb = {
+                label : linkObject.label,
+                url : linkObject.link,
+                id : linkObject.id
+            };
+            if (linkObject.type === 'collection' || linkObject.type === 'user'){
+                self.breadcrumbs([crumb]);
+                return;
+            }
+            if (linkObject.type === 'breadcrumb'){
+                self.breadcrumbs().splice(linkObject.index+1, self.breadcrumbs().length-linkObject.index-1);
```

Comment:
Sorry my bad, I updated the linkObject class to reflect this better. However I thought I was using the right method, splice is what I am trying to do.